src/ray/raylet/node_manager.cc

```diff
@@ -2169,10 +2169,14 @@ void NodeManager::SubmitTask(const Task &task, const Lineage &uncommitted_lineag
   RAY_LOG(DEBUG) << "Submitting task: " << task.DebugString();
 
   if (local_queues_.HasTask(task_id)) {
-    RAY_LOG(WARNING) << "Submitted task " << task_id
-                     << " is already queued and will not be restarted. This is most "
-                        "likely due to spurious reconstruction.";
-    return;
+    RAY_LOG(WARNING)
+        << "Submitted task " << task_id
+        << " is already queued, we will remove it from queue and "
+           "recreate the task, because the dispatch task callback of the old task "
+           "is invalid if the GCS server is restarted.";
```

Comment:
Just realize that we should only do this for actor creation tasks when gcs service is enabled.

Reply:
```
Submitted actor creation task ... is already queue. This is most likely due to a GCS restart. We will remove the old one from the queue, and enqueue the new one.
```

Reply:
fixed